.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,55 @@
+name: deploy
+
+on:
+  pull_request:
+  push:
+    branches: [main, '*-assessment-year']
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  publish-docker-image:
+    runs-on: ubuntu-latest
+    # Set token permissions so that we can publish to GitHub Container Registry
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          # Tag the following types of images:
+          #   * On a branch, tag with the branch name (e.g. `master`)
+          #   * On a PR, tag with the PR number (e.g. `pr-12`)
+          #   * On all events, tag with the short git SHA (e.g. `e956384`)
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=sha
```

Comment:
Does this seem like the correct tagging scheme to you?

One subtlety here is that images persist indefinitely in GHCR, so by tagging every image with its `sha` we run the risk of using a very large amount of storage for images that correspond to individual commits. This strikes me as a good thing for the main branches (`master` and `*-assessment-year`) since we want to be able to recover old models that were used on those branches, but I bet we could reduce the storage use by not producing SHA-tagged images on pull requests.

That being said, my read of the pricing for [GitHub Packages](https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages#about-billing-for-github-packages) suggests that storage is free and unlimited for public packages, so maybe we just push everything and don't worry too much about it. I'm not sure, curious what you think!

Reply:
**suggestion:** Instead of tagging and creating an image for every commit, let's create an image for each git tag. That way, when we tag `assessment-year-2024`, we'll make a long-lived image with the same name that has the dependencies necessary to run the 2024 model already built/bundled.